src/App.jsx

```diff
@@ -14,70 +15,91 @@ export const goods = [
   'Garlic',
 ];
 
-export const App = () => (
-  <main className="section container">
-    <h1 className="title is-flex is-align-items-center">No goods selected</h1>
-
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
-
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+export const App = () => {
+  const [selectedGood, setSelectedGood] = useState('Jam');
 
+  const ItemsList = ({ products }) => (
     <table className="table">
       <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+        {products.map(product => (
+          <Item
+            key={product}
+            product={product}
+          />
+        ))}
+      </tbody>
+    </table>
+  );
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+  const RemoveButton = () => (
+    <button
+      data-cy="RemoveButton"
+      type="button"
+      className="button is-info"
+      onClick={() => setSelectedGood('')}
+    >
+      -
+    </button>
+  );
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+  const Item = ({ product }) => {
+    const isSelected = selectedGood === product;
```

Comment:
Item as a component shouldn't know anything about selectedGood.
Also, naming isn't good, maybe GoodItem, the same about ItemsList